src/jitlayers.cpp

```diff
@@ -1629,6 +1629,10 @@ JuliaOJIT::JuliaOJIT()
     DLSymOpt(std::make_unique<DLSymOptimizer>(false)),
     ContextPool([](){
         auto ctx = std::make_unique<LLVMContext>();
+#ifndef JL_LLVM_OPAQUE_POINTERS
+        if (ctx.supportsTypedPointers())
+            ctx.setOpaquePointers(false);
```

Comment:
```suggestion
        if (ctx->supportsTypedPointers())
            ctx->setOpaquePointers(false);
```

I've made that change in https://github.com/JuliaLang/julia/pull/51840